src/System.Private.Interop/src/Internal/Runtime/CompilerHelpers/RuntimeInteropData.CoreRT.cs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Internal.Runtime.Augments;
+using Internal.NativeFormat;
+using Internal.Runtime.TypeLoader;
+using Internal.Reflection.Execution;
+using System.Runtime.InteropServices;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    internal partial class RuntimeInteropData
+    {
+        public override bool TryGetMarshallerDataForDelegate(RuntimeTypeHandle delegateTypeHandle, out McgPInvokeDelegateData data)
+        {
+            IntPtr openStub, closedStub, delegateCreationStub;
+            if (!TryGetMarshallersForDelegate(delegateTypeHandle, out openStub, out closedStub, out delegateCreationStub))
+            {
+                data = default(McgPInvokeDelegateData);
+                return false;
+            }
+
+            data = new global::System.Runtime.InteropServices.McgPInvokeDelegateData()
+            {
+                ReverseOpenStaticDelegateStub = openStub,
+                ReverseStub = closedStub,
```

Comment:
This was done in ProjectN for performance reasons. The open static version is more performant compared to the closed one, because we can directly call the static function.